### PR TITLE
[docs] add radio defgroup for proper documentation

### PR DESCRIPTION
**Pull request by gavinmcgovern**

### CODE REVIEW

**Comment by Vyrastas** [comment]
We forgot to add this via https://github.com/openthread/openthread/pull/9071

**Review by Vyrastas — APPROVED** [review]

File: doc/ot_api_doc.h [hunk 1]
```diff
@@ -142,6 +142,7 @@
  * @defgroup api-mesh-diag            Mesh Diagnostics
  * @defgroup api-ncp                  Network Co-Processor
  * @defgroup api-network-time         Network Time Synchronization
+ * @defgroup api-radio-stats          Radio Statistics
```
Comment (abtink) [inline doc/ot_api_doc.h:145]: LGTM.
One suggestion: How about if we rename and use `api-radio`? So it is more generic and can potentially be used for any future `otRadio` APIs?

Comment (Vyrastas) [reply doc/ot_api_doc.h:145]: I'm fine with that, @gavinmcgovern if you can update the source radio-stats.h file to align with that as well

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

**Comment by Vyrastas** [comment]
@gavinmcgovern Will need to squash/rebase this

**Review by jwhui — APPROVED** [review]

**Pull request merged**